internal/controller/pipelinegate_controller.go

```diff
@@ -237,15 +237,20 @@ func (r *PipelineGateReconciler) updateJob(ctx context.Context, reconciledJob *b
 	return nil
 }
 
-func (r *PipelineGateReconciler) updateConsoleGate(gate *v1alpha1.PipelineGate) error {
-	updateAttrs, err := gate.Status.GateUpdateAttributes()
+func (r *PipelineGateReconciler) updateConsoleGate(gate *v1alpha1.PipelineGate, cachedGate *console.PipelineGateFragment) error {
+	sha, err := utils.HashObject(gateUpdateAttributes(cachedGate))
```

Comment:
why not use `gate.Status.GateUpdateAttributes()` here to avoid the possibility of drift between the impl of both those functions?

Reply:
gate Status reflects the current state (rather final state because this method is called in defer function after all operations) and I have to compare it with the API state (cachedGate) to decide if I should save new API attributes.

Reply:
Ah ok as long as it works that's fine w/ me.